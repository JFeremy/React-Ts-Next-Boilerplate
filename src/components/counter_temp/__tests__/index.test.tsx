import { fireEvent, render, screen } from '@testing-library/react';

import CounterTemp from '..';

describe('CounterTemp', () => {
  const expectedTexts = {
    increment: 'Lorem',
    decrement: 'Ipsum',
  };

  it('shoud have expected texts and buttons', async () => {
    render(
      <CounterTemp
        increment={expectedTexts.increment}
        decrement={expectedTexts.decrement}
      />,
    );

    const buttons = screen.getAllByRole('button');

    expect(screen.getByText(expectedTexts.increment)).toBeInTheDocument();
    expect(screen.getByText(expectedTexts.decrement)).toBeInTheDocument();
    expect(screen.getByText(0)).toBeInTheDocument();
    expect(buttons).toHaveLength(2);
  });

  it('shoud count value be 1 after increment button clicked', async () => {
    render(
      <CounterTemp
        increment={expectedTexts.increment}
        decrement={expectedTexts.decrement}
      />,
    );

    fireEvent.click(screen.getByText(expectedTexts.increment));

    expect(screen.getByText(1)).toBeInTheDocument();
  });

  test('shoud count value be -1 after des button clicked', async () => {
    render(
      <CounterTemp
        increment={expectedTexts.increment}
        decrement={expectedTexts.decrement}
      />,
    );

    fireEvent.click(screen.getByText(expectedTexts.decrement));

    expect(screen.getByText(-1)).toBeInTheDocument();
  });

  it('component matches snapshot', async () => {
    const view = render(
      <CounterTemp
        increment={expectedTexts.increment}
        decrement={expectedTexts.decrement}
      />,
    );
    expect(view.container).toMatchSnapshot();
  });
});
