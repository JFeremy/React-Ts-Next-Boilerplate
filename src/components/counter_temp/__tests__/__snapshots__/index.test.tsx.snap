// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterTemp component matches snapshot 1`] = `
<div>
  <div
    class="d_flex items_center justify_center gap_5 flex_row"
  >
    <button
      class="text_red text_center bg_gray.800 rounded_2 fs_1rem px_10px py_5px"
    >
      Ipsum
    </button>
    <p>
      0
    </p>
    <button
      class="text_red text_center bg_gray.800 rounded_2 fs_1rem px_10px py_5px"
    >
      Lorem
    </button>
  </div>
</div>
`;
